.rubocop.yml

```diff
@@ -79,3 +79,7 @@ Style/GuardClause:
 # Case equality is not intrinsically problematic.
 Style/CaseEquality:
   Enabled: false
+
+# Sometimes assignment in conditionals is the right thing to do
+Lint/AssignmentInCondition:
+  Enabled: false
```

Comment:
If I recall correctly, you can add parenthesis around your assignment and this cop won't bother you:
```ruby
if (i = path.rindex('::'))
```